### ISSUE TITLE
Replace deprecated Boston housing prices dataset with California housing dataset

### ISSUE THREAD

**Issue by nzw0301** [optuna/optuna#2971]
## Motivation

Since scikit-learn 1.0.0, the Boston housing prices dataset has been deprecated due to the ethical problem. The dataset will be removed at scikit-learn 1.2.0. Please see also [the scikit-learn's page](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html).

The scikit-learn documentation suggesting using [California housing dataset](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html#sklearn.datasets.fetch_california_housing
) instead of the Boston dataset.

## Description

Please replace `load_boston` with `fetch_california_housing`.

## Additional context (optional)
Optuna has not used scikit-learn >= 1.0.0. yet.
See relased issue: https://github.com/optuna/optuna/pull/2953.

**Comment by nzw0301** [comment]
Note that according to the [git blame of `fetch_califolnia_housing`](https://github.com/scikit-learn/scikit-learn/blame/844b4be24d20fc42cc13b957374c718956a0db39/sklearn/datasets/_california_housing.py#L53), this is dataset is relatively old, so the backward compatibility seems fine; older scikit-learn with optuna works well.

**Comment by avats-dev** [comment]
Hii @nzw0301, I want to contribute to optuna and I have long term aims to contribute here. I'm thinking of starting with this one. 
I understood that I have to replace boston housing dataset with california housing dataset as mentioned in the deprecation warning from scikit-learn and that california housing dataset is also backward compatible upto almost ver-0.2 of scikit-learn. 

So how should I proceed further?

**Comment by nzw0301** [comment]
Hi @avats-dev, thank you for your comment! 

> So how should I proceed further? Should I replace every instance of load_boston in the repo?

You are right. I think the pull request replace all `load_boston` with `fetch_califolnia_housing`.

**Comment by avats-dev** [comment]
Ok, working on it! Would get back if in need of some help. Thanks!

**Comment by nzw0301** [comment]
Brilliant! I'm looking forward to your contribution 🤞🏻

**Comment by KapilM26** [comment]
Hi, if this is not done yet, I would like to pick it up.

**Comment by nzw0301** [comment]
@KapilM26 thank you for your comment! But I suppose that @avats-dev is working on this issue.

**Comment by avats-dev** [comment]
Yeah, @KapilM26 I'm working on this and would open a PR soon. I was busy last some of the days that's why I couldn't do this yet.

**Comment by nzw0301** [comment]
Great. Thank you for letting me know about the unlabelled issue. I've added the `Hacktoverfest` label to this issue.
